packages/number-field/src/vaadin-number-field.js

```diff
@@ -234,6 +221,42 @@ export class NumberField extends InputFieldMixin(SlotStylesMixin(ThemableMixin(E
       })
     );
     this.addController(new LabelledInputController(this.inputElement, this._labelNode));
+
+    this.inputElement.addEventListener('wheel', (e) => this._onWheel(e));
+  }
+
+  /**
+   * The `inputmode` attribute influences the software keyboard that is shown on touch devices.
+   * https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/inputmode
+   * Browsers and operating systems are quite inconsistent about what keys are available.
+   * We choose between numeric and decimal based on whether negative numbers are allowed,
+   * and based on testing on various devices to determine what keys are available.
+   * @private
+   */
+  __setInputMode(input, min) {
+    const hasNegative = isNaN(min) || min < 0;
```

Comment:
I wonder if changing the keyboard layout based on whether negative values are allowed is really worth it. I can image these layouts changing at some point, or being different between:
- mobile OS versions
- browsers
- custom keyboards that the user has installed

and this logic diverging between Vaadin versions. Looking at the code right now, I'm pretty sure that there are cases that this doesn't cover. It might be better to keep this as dumb as possible, rather than implying that we have a solid keyboard layout detection feature here.

Same goes for decimal vs. numeric.

Also setting inputmode to `text` for iPhones seems questionable. That optimizes the whole thing for users who want to enter negative numbers, rather than those who want to enter positive numbers (I assume that a user can still switch the layout to get a minus character somehow).

In short, I would suggest to just use a single default keyboard layout, regardless whether negative or decimal numbers are allowed. Ideally developers can still override it if they want to attempt to implement some logic like the one that is suggested by this PR.

Reply:
Thank you for detailed review. I will remove relevant changes from this PR and use `numeric` for now.
Created #3100 so we can discuss this in that issue (we might also need to do some manual testing).